### PR TITLE
Eviction service bug fixes and improvements

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/common/rest/TitusExceptionMapper.java [hunk 1]
```diff
@@ -232,6 +236,34 @@ private Response fromJobManagerException(JobManagerException e) {
...
+
+        switch (e.getErrorCode()) {
+            case BadConfiguration:
+                errorBuilder.status(HttpServletResponse.SC_BAD_REQUEST);
+                break;
+            case CapacityGroupNotFound:
+                errorBuilder.status(HttpServletResponse.SC_NOT_FOUND);
+                break;
+            case TaskNotFound:
+                errorBuilder.status(HttpServletResponse.SC_NOT_FOUND);
```
Comment (corindwyer) [inline titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/common/rest/TitusExceptionMapper.java]: group not founds together?

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/AvailabilityPercentageLimitDisruptionBudgetPolicy.java [hunk 2]
```diff
@@ -17,9 +17,13 @@
...
 
 import java.util.Objects;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
 
 public class AvailabilityPercentageLimitDisruptionBudgetPolicy extends DisruptionBudgetPolicy {
 
+    @Min(0)
+    @Max(100)
     private final double percentageOfHealthyContainers;
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/AvailabilityPercentageLimitDisruptionBudgetPolicy.java]: Agree. It is basic validation only.

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/sanitizer/JobAssertions.java [hunk 3]
```diff
@@ -105,6 +105,19 @@ public boolean areEnvironmentVariablesNotTooLarge(Map<String, String> environmen
         return totalSize <= MAX_ENVIRONMENT_VARIABLE_SIZE_BYTES;
     }
 
+    public boolean isValidContainerHealthServiceName(String name) {
+        String[] validNames = configuration.getContainerHealthProviders().split(",");
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/sanitizer/JobAssertions.java]: Something like we have to regexp. I will look into this.

File: titus-server-master/src/main/java/com/netflix/titus/master/eviction/service/quota/TitusQuotasManager.java [hunk 4]
```diff
@@ -87,18 +89,29 @@ public void shutdown() {
...
-        }
-
-        if (systemQuotaController.getQuota() < 1 || jobQuotaController.getQuota() < 1) {
-            return false;
+            return UNKNOWN_JOB;
         }
 
         synchronized (lock) {
-            return systemQuotaController.consume(task.getId()) && jobQuotaController.consume(task.getId());
+            if (systemQuotaController.getQuota() < 1) {
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-server-master/src/main/java/com/netflix/titus/master/eviction/service/quota/TitusQuotasManager.java]: As discussed offline, I will simplify it to only call `consume`.

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/common/rest/TitusExceptionMapper.java (hunk 1 above)
Comment (tbak) [reply titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/common/rest/TitusExceptionMapper.java]: Good catch

**Pull request merged**